src/pages/create-product/index.js

```diff
@@ -0,0 +1,170 @@
+import BaseComponent from "../../components/base-component.js";
+import { createProduct, getBrands, getCategories } from "../../api/products.js";
+
+import "./create-product.css";
+
+export default class CreateProductPage extends BaseComponent {
+  constructor() {
+    super();
+    this.createProduct = createProduct;
+    this.getCategories = getCategories;
+    this.getBrands = getBrands;
+
+    this.init();
+    this.initEventListeners();
+    this.loadData();
+  }
+
+  async loadData() {
+    try {
+      const [categories, brands] = await Promise.all([
+        this.getCategories(),
+        this.getBrands(),
+      ]);
+
+      this.update(categories, brands);
+    } catch (error) {
+      this.showAlert("error", error.message);
+    }
+  }
+
+  update(categories = [], brands = []) {
+    this.updateOptions("brandsSelect", brands);
+    this.updateOptions("categoriesSelect", categories);
+
+    this.subElements.fieldset.removeAttribute("disabled");
+  }
+
+  updateOptions(name = "", data = []) {
+    const fragment = new DocumentFragment();
+
+    for (const brand of data) {
+      const key = brand.toLowerCase().split(" ").join("_");
+      const option = new Option(brand, key);
+      fragment.append(option);
+    }
+
+    this.subElements[name].replaceChildren(fragment);
```

Comment:
**Scalability Issue**: The method updateOptions manipulates the DOM within a loop, which can cause reflows and repaints on each iteration. This is inefficient and can significantly impact performance and scalability, especially with a large number of options. <br> **Fix**: Modify the updateOptions method to build the entire HTML content as a string or use DocumentFragment and update the DOM once outside the loop to minimize reflows and repaints. <br> **Code Suggestion**: 
 ```
 -    for (const brand of data) {
 -      const key = brand.toLowerCase().split(" ").join("_");
 -      const option = new Option(brand, key);
 -      fragment.append(option);
 +    const optionsHtml = data.map(brand => {
 +      const key = brand.toLowerCase().split(" ").join("_");
 +      return '<option value="${key}">${brand}</option>';
 +    }).join('');
 +    this.subElements[name].innerHTML = optionsHtml;
 ```